ambari-server/src/main/python/ambari_server/setupSecurity.py

```diff
@@ -274,6 +275,56 @@ def __init__(self, options):
   def no_ldap_sync_options_set(self):
     return not self.ldap_sync_all and not self.ldap_sync_existing and self.ldap_sync_users is None and self.ldap_sync_groups is None
 
+def getLdapPropertyFromDB(properties, admin_login, admin_password, property_name):
+  ldapProperty = None
+  url = get_ambari_server_api_base(properties) + SETUP_LDAP_CONFIG_URL
+  admin_auth = base64.encodestring('%s:%s' % (admin_login, admin_password)).replace('\n', '')
+  request = urllib2.Request(url)
+  request.add_header('Authorization', 'Basic %s' % admin_auth)
+  request.add_header('X-Requested-By', 'ambari')
+  request.get_method = lambda: 'GET'
+  request_in_progress = True
+
+  sys.stdout.write('\nFetching LDAP configuration from DB')
+  numOfTries = 0
+  while request_in_progress:
+    numOfTries += 1
+    if (numOfTries == 60):
+      raise FatalException(1, "Could not fetch LDAP configuration within a minute; giving up!")
+    sys.stdout.write('.')
+    sys.stdout.flush()
+
+    try:
+      response = urllib2.urlopen(request)
```

Comment:
The response object is not closed.

You can use:

```python
from contextlib import closing
with closing(urllib2.urlopen(request)) as response:
  ...
  response.read()
  ...
```

Reply:
Thanks for this catch; it's good to learn these kind of stuff.

Reply:
Fixed